packages/cockpit/src/balance/operations/shared.js

```diff
@@ -0,0 +1,120 @@
+import {
+  __,
+  always,
+  apply,
+  applySpec,
+  assoc,
+  both,
+  cond,
+  either,
+  has,
+  ifElse,
+  isEmpty,
+  isNil,
+  juxt,
+  lt,
+  map,
+  path,
+  pathEq,
+  pathOr,
+  pipe,
+  pluck,
+  prop,
+  propEq,
+  propSatisfies,
+  subtract,
+  sum,
+  T,
+  when,
+} from 'ramda'
+
+export const transformMovementTypePropTo = (propPath, to) => pipe(
+  path(propPath),
+  when(either(isNil, isEmpty), always(0)),
+  Math.abs,
+  assoc('amount', __, { type: to })
+)
+
+const lessThanZero = lt(__, 0)
+
+export const isNegative = propSatisfies(lessThanZero)
+
+export const buildNetAmount = (buildOutcoming, buildOutgoing) => pipe(
+  juxt([
+    pipe(buildOutcoming, pluck('amount'), sum),
+    pipe(buildOutgoing, pluck('amount'), sum),
+  ]),
+  apply(subtract)
+)
+
+export const isNilProp = type => propSatisfies(isNil, type)
+
+export const getOperationDate = (dateType, fallbackDateType) => pipe(
+  when(has('movement_object'), prop('movement_object')),
+  ifElse(
+    isNilProp(dateType),
+    prop(fallbackDateType),
+    prop(dateType)
+  ),
+  when(
+    either(isNil, isEmpty),
+    always(null)
+  )
+)
+
+export const getSourceId = ifElse(
+  pathEq(['movement_object', 'type'], 'inter_recipient'),
+  path(['movement_object', 'source_id']),
+  always(null)
+)
+
+export const getTarget = ifElse(
```

Comment:
```suggestion
export const getTargetId = ifElse(
```
seria legal atualizar a nomenclatura  para manter o mesmo padrão de `getSourceId` e `getTransactionId`